### PR TITLE
[Dev] Fix compilation issues related to MSVC and Windows.h

### LINKED ISSUE #5368: DuckDB 0.6.0 on Windows (MSVC) Compile Error
DuckDB 0.6.0 on Windows (MSVC) Compile Error:
.../test/sql/storage/test_buffer_manager.cpp(263): error C2660: 'std::align': function does not take 1 arguments

Here is 0.6.0 version of test_buffer_manager.cpp (error):
D_ASSERT(buffer_manager.GetUsedMemory() == align(requested_size + Storage::BLOCK_HEADER_SIZE));

Here is 0.5.1 version of test_buffer_manager.cpp (without error):
D_ASSERT(buffer_manager.GetUsedMemory() == requested_size + Storage::BLOCK_HEADER_SIZE);

BTW: DuckDB not available via VCPKG...

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes #5368 

1. Don't do `using namespace std;` because of all the collisions it (could) cause(s), namely `align` was a collision this fixes
2. Include `common/windows.hpp` in `types.hpp` to try to avoid an issue with `max` defined by windows

The latter "fix" likely won't fix anything, but it's a thin header that does nothing if the target isn't windows, so it's not hurting anyone

### CODE REVIEW

File: test/sql/storage/test_buffer_manager.cpp [hunk 1]
```diff
@@ -5,8 +5,8 @@
 #include "test_helpers.hpp"
 #include "duckdb/main/client_context.hpp"
 
+// Can't use 'using namespace std' here because of conflicting names
```
Comment (Mytherin) [inline test/sql/storage/test_buffer_manager.cpp:8]: Can't we rather rename the "align" function so the conflict is removed? That seems less error-prone.

File: src/include/duckdb/common/types.hpp [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "duckdb/common/constants.hpp"
 #include "duckdb/common/single_thread_ptr.hpp"
 #include "duckdb/common/vector.hpp"
+#include "duckdb/common/windows.hpp"
```
Comment (Mytherin) [inline src/include/duckdb/common/types.hpp:15]: The `types.hpp` header gets included in almost every single file of the system - and `windows.h` is not particularly light on Windows. Can't we move `infinity` and `ninfinity` functions out of the header (as non-inline functions) and have them use `NumericLimits` instead?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Some comments below. Could we perhaps also add a C++ test that includes `<windows.h>` (**not** our cleaned up version in `duckdb/common/windows.hpp`) before including `duckdb.hpp` to verify that that works/compiles?

File: test/sql/storage/test_buffer_manager.cpp (hunk 1 above)
Comment (Tishj) [reply test/sql/storage/test_buffer_manager.cpp:8]: Yea I also thought about that, but isn't `using namespace std` just bad practice in general?
It seems we don't even need it in this case, as removing it did not mess at all with the compilation (though that could be related to clang + compiler arguments, so I'm waiting on CI to see if that holds true on every platform)

**Comment by Tishj** [comment]
Both `date_t::infinity` and `timestamp_t::infinity` are causing compilation issues in the ICU extension
which doesn't make that much sense, because the ICU extension is allowed to depend on core, no?
the functions are now defined in `types.cpp` - but it seems the ICU extension does not include the duckdb library when compiling? At least on windows

And these regressions are also interesting, because they don't seem unrelated, seeing as it involves a timestamp
but an infinity check not being inlined shouldn't cause a 10million timestamps -> strings cast to go from 0.39 -> 0.5 seconds

**Comment by Mytherin** [comment]
> the functions are now defined in types.cpp - but it seems the ICU extension does not include the duckdb library when compiling? At least on windows

All non-inlined functions used by extensions have to be marked with `DUCKDB_API` on Windows - otherwise linker errors pop up. That has to do with how the linking between the extensions and the system is performed.

> And these regressions are also interesting, because they don't seem unrelated, seeing as it involves a timestamp
but an infinity check not being inlined shouldn't cause a 10million timestamps -> strings cast to go from 0.39 -> 0.5 seconds

That's not too surprising - function calls are relatively expensive if you have to do them millions of times, that's why inlining is such a powerful optimization. Perhaps we should move these functions to the `Date` and `Timestamp` classes instead and make them inlined again?

**Comment by Tishj** [comment]
Ah that makes perfect sense (the `__cdecl` part of the symbol it's looking for should have given it away, looking back at it), I'll fix it, thanks for the quick response

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**